src/StatusQ/Core/StatusAssetSettings.qml

```diff
@@ -4,22 +4,35 @@ import StatusQ.Core 0.1
 QtObject {
     id: root
 
+    //icon
     property string name
+    property url source
     property real width
     property real height
     property color color
     property color disabledColor
-    property url source
     property int rotation
     property bool isLetterIdenticon
     property int letterSize: emoji ? 11 : (charactersLen == 1 ? _oneLetterSize : _twoLettersSize)
     property int charactersLen: 1
     property string emoji
     property string emojiSize: _emojiSize
-    property StatusIconBackgroundSettings background: StatusIconBackgroundSettings {}
 
     // can't have QtObject { id: d } inside QtObject, using '_' to indicate private
     readonly property int _oneLetterSize: Math.max(15, root.width / 2.0)
     readonly property int _twoLettersSize: Math.max(12, root.width / 2.2)
     readonly property string _emojiSize: "%1x%1".arg(Math.max(16, root.width * 0.6))
+
+    //icon bg
+    property real bgWidth
+    property real bgHeight
+    property int bgRadius
+    property color bgColor: "transparent"
+
+    //image
+    property real imgWidth
+    property real imgHeight
+    property url imgSource
+    property int imgStatus
```

Comment:
Just curious what's the purpose of this prop?

Reply:
I assume it's the state in which it is.
We have usages of images that are being loaded asynchronously and that can fail. 

Maybe it's something else though

Reply:
I have checked only in StatusQ and is used in StatusSmartIdenticon to load the letteridenticon instead if the image is not loaded because of an error. probably we could remove it, but we need to check and test in desktop as well.

Reply:
If we remove this, then we'll lose the fallback, which I don't think we want.
Especially with use cases like the discord import tool, chances are getting higher that there'll be assets that can't be loaded.

Reply:
Ok, makes sense but then we could simply add the default fallback as:
```
property real imgWidth: width
```
and so on